drasyl-node/src/main/java/org/drasyl/channel/ConnectionChannelInitializer.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2020-2023 Heiko Bornholdt and Kevin Röbert
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+ * OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.drasyl.channel;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.handler.codec.LengthFieldBasedFrameDecoder;
+import io.netty.handler.codec.LengthFieldPrepender;
+import org.drasyl.handler.connection.ConnectionAnalyzeHandler;
+import org.drasyl.handler.connection.ConnectionClosing;
+import org.drasyl.handler.connection.ConnectionConfig;
+import org.drasyl.handler.connection.ConnectionException;
+import org.drasyl.handler.connection.ConnectionHandler;
+import org.drasyl.handler.connection.ConnectionHandshakeCompleted;
+import org.drasyl.handler.connection.SegmentCodec;
+import org.drasyl.util.internal.UnstableApi;
+
+import static java.util.Objects.requireNonNull;
+
+@UnstableApi
+public abstract class ConnectionChannelInitializer extends ChannelInitializer<DrasylChannel> {
+    protected final ConnectionConfig config;
+
+    protected ConnectionChannelInitializer(final ConnectionConfig config) {
+        this.config = requireNonNull(config);
+    }
+
+    protected ConnectionChannelInitializer() {
+        this(ConnectionConfig.newBuilder().build());
+    }
+
+    @SuppressWarnings("java:S1188")
+    @Override
+    protected void initChannel(final DrasylChannel ch) throws Exception {
+        final ChannelPipeline p = ch.pipeline();
+
+        p.addLast(new SegmentCodec());
+        p.addLast(new ConnectionHandler(config));
+        p.addLast(new ConnectionAnalyzeHandler());
```

Comment:
Dieser Handler sollte Optional sein

Reply:
Ja kommt raus, sobald der Bumms läuft und dann ready-to-merge ist.